internal/providers/pluginfw/resources/notificationdestinations/data_notification_destinations.go

```diff
@@ -0,0 +1,126 @@
+	Client *common.DatabricksClient
+}
+
+type NotificationDestinationsInfo struct {
+	Id                       types.String                                     `tfsdk:"id" tf:"computed"`
+	DisplayNameContains      types.String                                     `tfsdk:"display_name_contains" tf:"optional"`
+	Type                     types.String                                     `tfsdk:"type" tf:"optional"`
+	NotificationDestinations []settings_tf.ListNotificationDestinationsResult `tfsdk:"notification_destinations" tf:"computed"`
+}
+
+func (d *NotificationDestinationsDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "databricks_notification_destinations_pluginframework"
+}
+
+func (d *NotificationDestinationsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: tfschema.DataSourceStructToSchemaMap(NotificationDestinationsInfo{}, nil),
+	}
+}
+
+func (d *NotificationDestinationsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if d.Client == nil {
+		d.Client = pluginfwcommon.ConfigureDataSource(req, resp)
+	}
+}
+
+func validateType(notificationType string) diag.Diagnostics {
+	validTypes := map[string]struct{}{
+		string(settings.DestinationTypeEmail):          {},
+		string(settings.DestinationTypeMicrosoftTeams): {},
+		string(settings.DestinationTypePagerduty):      {},
+		string(settings.DestinationTypeSlack):          {},
+		string(settings.DestinationTypeWebhook):        {},
+	}
+
+	if _, valid := validTypes[notificationType]; !valid {
+		return diag.Diagnostics{diag.NewErrorDiagnostic(fmt.Sprintf("Invalid type '%s'; valid types are EMAIL, MICROSOFT_TEAMS, PAGERDUTY, SLACK, WEBHOOK.", notificationType), "")}
+	}
+	return nil
+}
+
+func validateLength(destinations []settings_tf.ListNotificationDestinationsResult) diag.Diagnostics {
+	if len(destinations) == 0 {
+		return diag.Diagnostics{diag.NewErrorDiagnostic("Could not find any notification destinations with the specified criteria.", "")}
+	}
+	return nil
+}
+
+func appendToResponse(resp *datasource.ReadResponse, diags diag.Diagnostics) bool {
+	resp.Diagnostics.Append(diags...)
+	return resp.Diagnostics.HasError()
+}
+
+func (d *NotificationDestinationsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	w, diags := d.Client.GetWorkspaceClient()
+	if appendToResponse(resp, diags) {
+		return
+	}
+
+	var notificationInfo NotificationDestinationsInfo
+	if appendToResponse(resp, req.Config.Get(ctx, &notificationInfo)) {
+		return
+	}
+
+	notificationType := notificationInfo.Type.ValueString()
+	notificationDisplayName := notificationInfo.DisplayNameContains.ValueString()
+
+	if notificationType != "" {
+		if appendToResponse(resp, validateType(notificationType)) {
```

Comment:
we can join that condition to `if notificationType != "" && appendToResponse(resp, validateType(notificationType)) ...`